ouroboros-network/src/Ouroboros/Network/Point.hs

```diff
@@ -27,5 +30,20 @@ data Block slot hash = Block
   }
   deriving (Eq, Ord, Show, Generic, NoThunks)
 
+instance (FromCBOR slot, FromCBOR hash) => FromCBOR (Block slot hash) where
+  fromCBOR = do
+    decodeListLenOf 2
+    slot <- fromCBOR
+    Block slot <$> fromCBOR
+
+instance (ToCBOR slot, ToCBOR hash) => ToCBOR (Block slot hash) where
+  toCBOR blk = mconcat
+    [ encodeListLen 2
+    , toCBOR (blockPointSlot blk)
+    , toCBOR (blockPointHash blk)
+    ]
+
+deriving instance (Serialise slot, Serialise hash) => Serialise (Block slot hash)
+
```

Comment:
What is this used for?

Reply:
The `Block` type is a substructure of `Point`, so `Point` isn't serialisable without the above instances.

Reply:
Not sure what it is, but I removed the `ToCBOR` and `FromCBOR` instances and it compiles fine now.  :/

Reply:
I think it's because some compiler errors suggested I needed to add one when adding a `Serialise`, but that I eventually added an `Serialise` instance with an implementation that doesn't need it.